app/admin/views.py

```diff
@@ -201,20 +202,25 @@ def destinations_show(uuid):
     )
 
     if edit_form.validate_on_submit():
-        dest.name = edit_form.name.data
-        dest.address = edit_form.address.data
-        dest.point = 'SRID=4326;POINT({} {})'.format(
-            edit_form.destination_lon.data,
-            edit_form.destination_lat.data
-        )
+        if edit_form.submit.data:
```

Comment:
I'd prefer to not have form buttons do page navigation. Instead of checking for which button was pressed in the view here, use an `<a href>` in the template to implement the cancel action.